@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -236,6 +236,12 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
             } else {
                 // empty tiles are traditionally ignored and don't need decode
                 tile.forceHasGeometry(true);
```

Comment:
The call "forceHasGeometry(true)" should go after the "if (tileLoader.decodedTile !== undefined) {}" and be:

// Force to be handled as if tile has geometry, in case nothing has been extracted from the decodedTile.
tile.forceHasGeometry(tileLoader.decodedTile === undefined || !tile.hasGeometry());

If done like this, the "force" flag is only set if there is really no geometry, which may still be the case.